### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
**Description**

This PR addresses misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/jwst/commit/7c790943753ff8e6a250fef763b73b4b2af2a8a4#commitcomment-62574109

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/jwst/commit/b51a3ab9fa0f9d4138d00ffd5bfeaf4f7db7c5a6

Checklist
- [ ] Tests
- [ ] Documentation
- [ ] Change log
- [x] Milestone
- [x] Label(s)

I understand this PR is quite large. I regularly make PRs of this scale. I can split them by directory, by file type, by content form (function/comment/doc/etc.).

It will take me a bit, but I will try to annotate the changes based on observations. Fwiw, afaict the 3 test failures I encountered here are present in master.

Fwiw check-spelling is running on [fprime](https://github.com/nasa/fprime/actions/workflows/spelling.yml)

### CODE REVIEW

File: jwst/master_background/master_background_nrs_slits_step.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def _calc_master_background(self, data, user_background=None):
             mb_multislit = nirspec_utils.map_to_science_slits(pre_calibrated, master_background)
 
             # Now that the master background is pretending to be science,
-            # walk backwards through the steps to uncalibrate, using the
+            # walk backwards through the steps to decalibrate, using the
```
Comment (jsoref) [inline jwst/master_background/master_background_nrs_slits_step.py:257]: This code region has `decalibrate`, `uncalibrate`, and `pre-calibrated`. The first and last of these are in dictionaries. I can understand `uncalibrated` (which is in the dictionary). But I personally had trouble with the idea of actively uncalibrating (not in the dictionary).

Happy to drop. But I'd also like to learn about these.

File: jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml [hunk 2]
```diff
@@ -8,11 +8,11 @@ title: >
 description: |
   This model does all transforms through the NIRSpec prism:
   - computes the refraction index as a function of lambda.
-  - Applies Snell's law through fromt surface.
+  - Applies Snell's law through from surface.
```
Comment (jsoref) [inline jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml:11]: `from the`?

**Review by jsoref — COMMENTED** [review]
I did try to review everything (including test failures) before making this PR.

Initial corrections were suggested by Google Sheets. I made some overrides.

The tooling I'm uses sometimes does substring matching and sometimes does fixed string matches. The substring matches handle cases like `\tobstart`. And sometimes it fails which is why fixing `exact` broke something.

I will try to queue fixes to address the comments I've made here (mostly `exact` and some indentation), but I don't think I'll push them until I get feedback.

File: jwst/master_background/master_background_nrs_slits_step.py (hunk 1 above)
Comment (hbushouse) [reply jwst/master_background/master_background_nrs_slits_step.py:257]: We should stick with the original "uncalibrate" here.

File: jwst/mrs_imatch/mrs_imatch_step.py [hunk 3]
```diff
@@ -318,7 +318,7 @@ def _match_models(models, channel, degree, center=None, center_cs='image'):
         sigma_data.append(sigmas)
 
     # leaving in below commented out lines for
-    # Mihia to de-bug step when coefficients are NAN
+    # Mihai to de-bug step when coefficients are NAN
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/mrs_imatch/mrs_imatch_step.py:321]: The new spelling is the correct form for @mcara

File: jwst/regtest/test_infrastructure.py [hunk 4]
```diff
@@ -84,7 +84,7 @@ def test_diff_astropy_tables_columns(diff_astropy_tables, two_tables):
         assert diff_astropy_tables(path1, path2)
 
 
-@pytest.mark.xfail(reason='table meta comparison currently deactivated')
+@pytest.mark.xfail(reason='table meta comparison currently dexactivated')
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/regtest/test_infrastructure.py:87]: Agree. This should be dropped (original is correct).

File: jwst/rscd/rscd_sub.py [hunk 5]
```diff
@@ -114,14 +114,14 @@ def correction_decay_function(input_model, param):
     Applies rscd correction to science arrays
     The last frame value from the previous integration is calculated two ways:
     1. using second and third to last frames to extrapolated to the last frame
-    2. using the non saturating data, fit the data and expolate to last frame
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/rscd/rscd_sub.py:117]: Should be "extrapolate"

File: jwst/stpipe/tests/steps/stepwithmodel_hook.cfg [hunk 6]
```diff
@@ -1,4 +1,4 @@
-name = 'StepWithModleHook'
+name = 'StepWithModuleHook'
```
Comment (hbushouse) [inline jwst/stpipe/tests/steps/stepwithmodel_hook.cfg:1]: This should be "StepWithModelHook"

File: jwst/stpipe/tests/steps/stepwithmodel_hookreturn.cfg [hunk 7]
```diff
@@ -1,4 +1,4 @@
-name = 'StepWithModleHook'
+name = 'StepWithModuleHook'
```
Comment (hbushouse) [inline jwst/stpipe/tests/steps/stepwithmodel_hookreturn.cfg:1]: Same here. Should be "StepWithModelHook".

File: jwst/straylight/__init__.py [hunk 8]
```diff
@@ -1,4 +1,4 @@
 from .straylight_step import StraylightStep
-from .straylight import correct_mrs, correct_mrs_modshepard, shepard_2d_kernel
+from .straylight import correct_mrs, correct_mrs_modshephard, shephard_2d_kernel
```
Comment (hbushouse) [inline jwst/straylight/__init__.py:2]: No, original is correct.

File: jwst/straylight/__init__.py [hunk 9]
```diff
@@ -1,4 +1,4 @@
...
 
-__all__ = ['StraylightStep', 'correct_mrs', 'correct_mrs_modshepard', 'shepard_2d_kernel']
+__all__ = ['StraylightStep', 'correct_mrs', 'correct_mrs_modshephard', 'shephard_2d_kernel']
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/straylight/__init__.py:4]: Don't change.

File: jwst/straylight/straylight.py [hunk 10]
```diff
@@ -4,7 +4,7 @@
 # 1's for gaps between slices.
 #
 # there are two algorithms in the module
-# the default one that is applied is correct_mrs_modshepard
+# the default one that is applied is correct_mrs_modshephard
```
Comment (hbushouse) [inline jwst/straylight/straylight.py:7]: no

File: jwst/straylight/straylight.py [hunk 11]
```diff
@@ -170,14 +170,14 @@ def correct_mrs(input_model, slice_map):
     return output
 
 
-def correct_mrs_modshepard(input_model, slice_map, roi, power):
+def correct_mrs_modshephard(input_model, slice_map, roi, power):
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/straylight/straylight.py:173]: Don't change. Original is correct.

File: jwst/straylight/straylight.py [hunk 12]
```diff
@@ -170,14 +170,14 @@ def correct_mrs(input_model, slice_map):
...
     """
     Straylight correction using a modified Shepard algorithm.
 
     Corrects the MIRI MRS data for straylight using a Modified version of the
     Shepard algorithm. Straylight is determined using an inverse distance weighting
     function. The inverse distance weighting is determined by module
-    ``shepard_2d_kernel(roi,power)``.
+    ``shephard_2d_kernel(roi,power)``.
```
Comment (hbushouse) [inline jwst/straylight/straylight.py:180]: no

File: jwst/straylight/straylight.py [hunk 13]
```diff
@@ -263,7 +263,7 @@ def correct_mrs_modshepard(input_model, slice_map, roi, power):
     return output
 
 
-def shepard_2d_kernel(roi, power):
+def shephard_2d_kernel(roi, power):
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/straylight/straylight.py:266]: No to all instances of changing "shepard". Original is correct.

File: jwst/straylight/straylight_step.py [hunk 14]
```diff
@@ -86,10 +86,10 @@ def process(self, input):
                     self.log.info(' Modified Shepard weighting power %5.2f', self.power)
                     self.log.info(' Region of influence radius (pixels) %6.2f', self.roi)
 
-                    result = straylight.correct_mrs_modshepard(input_model,
-                                                               regions,
-                                                               self.roi,
-                                                               self.power)
+                    result = straylight.correct_mrs_modshephard(input_model,
```
Comment (hbushouse) [inline jwst/straylight/straylight_step.py:89]: no

File: jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/logical-1.0.0.yaml [hunk 15]
```diff
@@ -17,7 +17,7 @@ examples:
   -
     - |
         If the input is less that [1,2,3,4], set its value to [5,6,7,8].
-        The input array shoul dhave the same shape.
+        The input array should dhave the same shape.
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/logical-1.0.0.yaml:20]: Agree.

File: jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml (hunk 2 above)
Comment (hbushouse) [reply jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml:11]: "front"

File: jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml [hunk 16]
```diff
@@ -8,11 +8,11 @@ title: >
...
   - Rotates to back surface.
   - Applies reflection from back surface.
   - Rotates to front surface
-  - Applies Snell's law through fromt surface.
+  - Applies Snell's law through from surface.
```
Comment (hbushouse) [inline jwst/transforms/resources/schemas/stsci.edu/jwst_pipeline/snell-0.7.0.yaml:15]: front

**Review by hbushouse — COMMENTED** [review]
Wow, what an amazing amount of cleanup. I think 99% of it is OK, except where noted. Holding off on approval until others have had a chance to add their commentary and respond to a few questions were they are the experts.

File: jwst/datamodels/schemas/core.schema.yaml [hunk 17]
```diff
@@ -556,12 +556,12 @@ properties:
             fits_keyword: OPMODE
             blend_table: True
           gwa_xtilt:
-            title: Grating wheel tilt along instrmnt model Y axis
+            title: Grating wheel tilt along instrument model Y axis
```
[the remark this answers is not in the document]
Comment (jsoref) [reply jwst/datamodels/schemas/core.schema.yaml:559]: Ah, yes, I had caught some of the FITS items and discarded, I should have recognized this as being in that family.

File: jwst/associations/lib/rules_level2_base.py [hunk 18]
```diff
@@ -695,8 +695,8 @@ def _merge_asns(asns):
                             if member['expname'] in new_names
                         ]
                         current_product['members'].extend(new_members)
-                        merge_occured = True
-                if not merge_occured:
+                        merge_occurred = True
+                if not merge_occurred:
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/associations/lib/rules_level2_base.py:699]: Good to go.

File: jwst/associations/tests/test_level2_basics.py [hunk 19]
```diff
@@ -13,7 +13,7 @@
 )
 from jwst.associations.main import Main
 
-NONSSCIENCE = ['background']
+NONSCIENCE = ['background']
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/associations/tests/test_level2_basics.py:16]: Considering that this is not used anywhere else in the module, I say delete the whole line.

File: jwst/cube_build/cube_build_wcs_util.py [hunk 20]
```diff
@@ -15,7 +15,7 @@ def find_corners_MIRI(input, this_channel, instrument_info, coord_system):
     """ For MIRI channel data find the foot of this data on the sky
 
     For a specific channel on an exposure find the min and max of the
-    spatial coordinates, either in  alpha,beta or ra,dec dedpending
+    spatial coordinates, either in  alpha,beta or ra,dec depending
```
Comment (stscieisenhamer) [inline jwst/cube_build/cube_build_wcs_util.py:18]: Feel free to remove the extra space in front of `alpha` also.

File: jwst/engdblog/tests/test_engdblog.py [hunk 21]
```diff
@@ -55,7 +55,7 @@ def test_novalues(caplog, engdb):
 
 def test_all(caplog, engdb):
     mnemonic = 'INRSI_GWA_Y_TILT_AVGED'
-    cfg_file_name = 'engdblogste.cfg'
+    cfg_file_name = 'engdblogstep.cfg'
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/engdblog/tests/test_engdblog.py:58]: Good to go.

File: jwst/lib/set_telescope_pointing.py [hunk 22]
```diff
@@ -123,10 +123,10 @@ class Methods(Enum):
     #: Observatory orientation without velocity correction, TR 2021-05
     TR_202105 = ('tr_202105', 'calc_transforms_tr202105', 'calc_wcs_orig')
     #: Observatory orientation with velocity correction, TR 2021-05
-    TR_202105_VA = ('tr_202105_va', 'calc_transforms_velocity_abberation_tr202105', 'calc_wcs_orig')
-    #: TRACK and FINEGUIDE mode alorithm, TR version 2021-07
+    TR_202105_VA = ('tr_202105_va', 'calc_transforms_velocity_aberration_tr202105', 'calc_wcs_orig')
+    #: TRACK and FINEGUIDE mode algorithm, TR version 2021-07
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/set_telescope_pointing.py:127]: the corrected spelling is good. the corresponding change is made below, so all is consistent.

File: jwst/lib/set_telescope_pointing.py [hunk 23]
```diff
@@ -1364,7 +1364,7 @@ def calc_transforms_ops_tr_202111(t_pars: TransformParameters):
         )
 
 
-def calc_transforms_velocity_abberation_tr202105(t_pars: TransformParameters):
+def calc_transforms_velocity_aberration_tr202105(t_pars: TransformParameters):
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/set_telescope_pointing.py:1367]: as above, the correction is good.

File: jwst/lib/siafdb.py [hunk 24]
```diff
@@ -50,8 +50,8 @@
 SIAF.__new__.__defaults__ = (None, None, None, None, 0, 0, 3600, 3600,
                              (0, 1, 1, 0, 0, 0, 1, 1))
 
-SIAF_VERTICIES = ['XIdlVert1', 'XIdlVert2', 'XIdlVert3', 'XIdlVert4',
-                  'YIdlVert1', 'YIdlVert2', 'YIdlVert3', 'YIdlVert4']
+SIAF_VERTICES = ['XIdlVert1', 'XIdlVert2', 'XIdlVert3', 'XIdlVert4',
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/siafdb.py:53]: It is just used internally here. However, SDP may be using this also. I say it should remain unchanged until next build.

File: jwst/lib/siafdb.py [hunk 25]
```diff
@@ -204,10 +204,10 @@ def get_wcs(self, aperture, useafter):
         aperture = siaf[aperture.upper()]
 
         # Fill out the Siaf
-        verticies = tuple(getattr(aperture, key) for key in SIAF_VERTICIES)
+        vertices = tuple(getattr(aperture, key) for key in SIAF_VERTICES)
```
Comment (stscieisenhamer) [inline jwst/lib/siafdb.py:207]: as above, do not change SIAF_VERTICIES.

File: jwst/lib/tests/stp_v1.py [hunk 26]
```diff
@@ -306,19 +306,19 @@ def calc_wcs(v2ref, v3ref, v3idlyang, vidlparity,
...
 
-def get_pointing(obstart, obsend, result_type='first'):
+def get_pointing(obsstart, obsend, result_type='first'):
     """
     Get telescope pointing engineering data.
 
     Parameters
     ----------
-    obstart, obsend: float
+    obsstart, obsend: float
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/tests/stp_v1.py:315]: good to go.

File: jwst/lib/tests/test_set_telescope_pointing.py [hunk 27]
```diff
@@ -313,7 +313,7 @@ def test_change_engdb_url_fail():
         stp.get_pointing(
             Time('2019-06-03T17:25:40', format='isot').mjd,
             Time('2019-06-03T17:25:56', format='isot').mjd,
-            engdb_url='http://nonexistant.fake'
+            engdb_url='http://nonexistent.fake.example'
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/tests/test_set_telescope_pointing.py:316]: good to go.

File: docs/jwst/assign_wcs/reference_files.rst [hunk 28]
```diff
@@ -19,13 +19,13 @@ REFTYPE                                           Description
 :ref:`COLLIMATOR <collimator_reffile>`            NIRSpec Collimator Model                              NIRSpec
 :ref:`DISPERSER <disperser_reffile>`              Disperser parameters                                  NIRSpec
 :ref:`DISTORTION <distortion_reffile>`            Spatial distortion model                              FGS, MIRI, NIRCam, NIRISS
-:ref:`FILTEROFFSET <filteroffset_reffile>`        MIRI Imager fiter offsets                             MIRI, NIRCAM, NIRISS
+:ref:`FILTEROFFSET <filteroffset_reffile>`        MIRI Imager filter offsets                            MIRI, NIRCAM, NIRISS
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply docs/jwst/assign_wcs/reference_files.rst:22]: good to go.

**Review by stscieisenhamer — CHANGES_REQUESTED** [review]

File: jwst/assign_wcs/util.py [hunk 29]
```diff
@@ -1122,7 +1122,7 @@ def velocity_correction(velosys):
     """
     correction = (1 / (1 + velosys / c.value))
     model = astmodels.Identity(1) * astmodels.Const1D(correction, name="velocity_correction")
-    model.inverse = astmodels.Identity(1) / astmodels.Const1D(correction, name="inv_vel_correciton")
+    model.inverse = astmodels.Identity(1) / astmodels.Const1D(correction, name="inv_vel_correction")
```
[the remark this answers is not in the document]
Comment (nden) [reply jwst/assign_wcs/util.py:1125]: Change looks good, no impact.

File: jwst/lib/siafdb.py (hunk 25 above)
Comment (jsoref) [reply jwst/lib/siafdb.py]: Note: I've changed `vertices` but have left `SIAF_VERTICIES` unchanged.

**Comment by jsoref** [comment]
#6505 moved some code to spacetelescope/stcal#63

I don't think that repository is in my current set (I think I picked a top 7). There are >250 repositories in this org, and while I did 7 last night, I don't think there's enough time between now and new years for me to do all of them. If people like this PR, I'm happy to take requests for a second or third set of repositories to check.

**Comment by hbushouse** [comment]
The code moves in #6505 did not include any modules that are included in this PR, so we're clean.

**Comment by jsoref** [comment]
These two files used to be in this PR:
[jwst/dark_current/dark_class.py](https://github.com/jsoref/jwst/commit/b51a3ab9fa0f9d4138d00ffd5bfeaf4f7db7c5a6#diff-d81fe0160a6c470f5acc54e6f8f7673356b03a7f73a6cb843418e6b499bfbc7a)
[jwst/dark_current/dark_sub.py](https://github.com/jsoref/jwst/commit/b51a3ab9fa0f9d4138d00ffd5bfeaf4f7db7c5a6#diff-be61b26048be882f6eb8f44858de275e7181bfa7526029a6cbc243bc3b0f6e34)

https://github.com/spacetelescope/stcal/pull/63/files#diff-4e55d5694d0305463b692bf522797455d3b961c4e27bb3b07027fe50073b11f3R148
https://github.com/spacetelescope/stcal/pull/63/files#diff-4e55d5694d0305463b692bf522797455d3b961c4e27bb3b07027fe50073b11f3R284

It isn't a big deal. I'll catch the files eventually when I visit that repository.

I only made the comment to explain the need for the extra [force push](https://github.com/spacetelescope/jwst/pull/6524#event-5822685579) (which resolved the conflict from that PR being merged).

**Comment by stscieisenhamer** [comment]
@jsoref If ready for a re-review, please hit the the "re-request review" button next to the reviewer names. That will help signal when to go through things again. Thanks!

**Review by stscieisenhamer — APPROVED** [review]
LGTM

@hbushouse I'm clear.  Don't know about others.

**Review by nden — APPROVED** [review]

**Review by hbushouse — APPROVED** [review]
A final review indicates that all previous issues and questions have been fixed or determined to be correct as is. So I this all looks good now.

**Comment by hbushouse** [comment]
Failing CI tests are unrelated.

**Pull request merged**